source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -0,0 +1,794 @@
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Ellipsoidal chunk");
+        {
+          /**
+           * Get latitude and longitudes defining region of interest from
+           * the parameter file.
+           */
+          corners.resize(4);
+          std::string NEcorner = prm.get("NE corner");
+          std::string NWcorner = prm.get("NW corner");
+          std::string SWcorner = prm.get("SW corner");
+          std::string SEcorner = prm.get("SE corner");
+
+          /**
+           * make a list of what corners are present and check that there should be one or two corner missing,
+           * otherwise throw an exception
+           */
+          std::vector<bool> present(4,true);
+          unsigned int missing = 0;
+          if (NEcorner == "")
+            {
+              missing++;
+              present[0] = false;
+            }
+          if (NWcorner == "")
+            {
+              missing++;
+              present[1] = false;
+            }
+          if (SWcorner == "")
+            {
+              missing++;
+              present[2] = false;
+            }
+          if (SEcorner == "")
+            {
+              missing++;
+              present[3] = false;
+            }
+
+          AssertThrow (missing != 0,
+                       ExcMessage ("Only two or three of the four corners points should be provided."));
+          AssertThrow (missing == 1 || missing == 2,
+                       ExcMessage ("Please provide two or three corners points."));
+
+          corners[0] = Utilities::string_to_double(Utilities::split_string_list(NEcorner,':'));
+          corners[1] = Utilities::string_to_double(Utilities::split_string_list(NWcorner,':'));
+          corners[2] = Utilities::string_to_double(Utilities::split_string_list(SWcorner,':'));
+          corners[3] = Utilities::string_to_double(Utilities::split_string_list(SEcorner,':'));
+          rotation_longitude = corners[0][0];
+          rotation_latitude = corners[0][1];
+          bottomDepth = prm.get_double("Depth");
+          semi_major_axis_a = prm.get_double("Semi-major axis");
+          Eccentricity = prm.get_double("Eccentricity");
+          semi_minor_axis_b=std::sqrt((1 - pow(Eccentricity,2)) * pow(semi_major_axis_a,2));
+          EW_subdiv = prm.get_double("East-West subdivision");
+          NS_subdiv = prm.get_double("North-South subdivision");
+          depth_subdiv = prm.get_double("Depth subdivision");
+
+          // Check whether the corners of the rectangle are really place correctly
+          if (present[0] == true && present[1] == true)
+            {
+              AssertThrow (corners[0][0] <= corners[1][0],
+                           ExcMessage ("The longitude of the NE corner (" + std::to_string(corners[0][0]) + ") cannot be smaller then the longitude of the NW corner (" + std::to_string(corners[1][0]) + ")."));

```

Comment:
Hmm, I now notice that std::to_string is only available since c++11. Have to find alternative.